datumaro/plugins/mapillary_vistas_format/extractor.py

```diff
@@ -0,0 +1,280 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+import glob
+import json
+import logging as log
+import os
+import os.path as osp
+
+import numpy as np
+
+from datumaro.components.annotation import (
+    AnnotationType, CompiledMask, LabelCategories, Mask, MaskCategories,
+    Polygon,
+)
+from datumaro.components.extractor import DatasetItem, SourceExtractor
+from datumaro.util.image import Image, find_images, lazy_image, load_image
+from datumaro.util.mask_tools import bgr2index
+
+from .format import (
+    MapillaryVistasLabelMaps, MapillaryVistasPath, MapillaryVistasTask,
+    make_mapillary_instance_categories, parse_config_file,
+)
+
+
+class _MapillaryVistasExtractor(SourceExtractor):
```

Comment:
I see, it is the same as COCO panoptic. Can we reuse some code?

Reply:
Yes, you right. The format `mapillary_vistas_panoptic` very simillar with `coco_panoptic`.
But they have small differences:
    1) JSON files in `coco_panoptic` have numeric id for dataset items and we extract item id from path to the panoptic mask, but JSON files in `mapillary_vistas_panoptic` have text item ids and we use this field as item id in Datumaro`s representation of dataset.
    2) Also `coco_panoptic` doesn't have color field for categories in panoptic config, but `mapillaty_vistas_panoptic` has it.